recipes/curl/curl.inc

```diff
@@ -24,22 +24,30 @@ EXTRA_OECONF_IPV6:USE_curl_ipv6 = "--enable-ipv6"
 
 RECIPE_FLAGS += "curl_ssl"
 DEPENDS:>USE_curl_ssl = " libssl libcrypto"
+RDEPENDS_${PN}:>USE_curl_ssl = " libssl libcrypto"
+RDEPENDS_${PN}-dev:>USE_curl_ssl = " libssl-dev libcrypto-dev"
```

Comment:
This might not be necessary after all.  It looks like library.oeclass should add these lib*-dev dependencies based on the RDEPENDS_${PN} lib dependencies.  Could you try removing and verify with oe show that RDEPENDS_curl-dev still contains libssl-dev and libcrypto-dev ?